### ISSUE TITLE
Can't get success on login

### ISSUE THREAD

**Issue by KevinRM** [playgameservices/play-games-plugin-for-unity#1780]
Hi,

Can you help me with this please?

I install the plugin 0.9.39, configure correctly in google play console and Unity but authentication return false.

![captura](https://user-images.githubusercontent.com/10706601/26849429-109fb0d8-4afc-11e7-9645-e63900d41e2c.PNG)

This is my code, in google play console y have dissabled saved games, I don't need, but in the code, if uncoment EnableSavedGames the login proccess show the typical banner that indicate the user authenticated but the text in function "loginGoogle" print "No logueado: False". Then, if I comment the EnableSavedGames, run the game and call loginGoogle, print "Logueado: True", then, call other time to log out and print "Ya estaba logueado" and finally, call other time to login, select account but print "No logueado: False".

The unique form that login is uncomment EnableSavedGames but print "No logueado: False". I attach the log file

[logs.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/1056274/logs.txt)

**Comment by claywilkinson** [comment]
This line in your log file looks like you have a problem related to the aar files being included?
`
E/Parcel  ( 2343): Caused by: java.lang.NoClassDefFoundError: com/google/android/gms/auth/api/signin/internal/SignInConfiguration
`

Can you try running the jar resolver again and if it still does not work share the list of files in Assets/Plugins/Android ?

**Comment by KevinRM** [comment]
Thanks @claywilkinson for answer. I ran jar resolver (Assets - Play Services Resolver - Android Resolver - Resolver Client Jars) but I have the same problem.

This is my Plugin/Android directory

![captura](https://user-images.githubusercontent.com/10706601/26854770-840b68a6-4b0f-11e7-8997-67e4ff5942f4.PNG)

**Comment by claywilkinson** [comment]
Hmm, weird - What version of Android are you running on ?  Which version of Google Play Services is installed on that device?

**Comment by KevinRM** [comment]
@claywilkinson I test on Android 4.2.2, 4.3 and 6.0.1. In all, I have the same problem. The Google Play Services is the latest I think, updated on 26 May to 11.0.55

I don't know what happens :S

**Comment by KevinRM** [comment]
I've been trying different ways and got it working by commenting the line ".RequestIdToken()", but... This is necessary for Firebase authentication?

**Comment by claywilkinson** [comment]
Can you share the log of including RequestIdToken() and it failing?  Since it works OK if you comment this line out, we know the aar and other libraries are good.

One thing to confirm is that the web client id is from a web app that is linked to your game in the Play console.  This should be used in the setup dialog for this plugin.  If it is already there, you can try re-running setup which will regenerate Assets/GooglePlayGames/GameInfo.cs.

**Comment by KevinRM** [comment]
Ok, this is the log file with .RequestIdToken() [logs.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/1060713/logs.txt)

I check the web client id and try re-running setup but I have the same problem

**Comment by claywilkinson** [comment]
This log looks like you are not using 0.9.39a  Can you delete the Assets/GooglePlayGames and Assets/Plugins/Android/MainLibProj and reimport 0.9.39a?

**Comment by KevinRM** [comment]
Ohh sorry, I hadn't seen the new version. I delete and reimport the new but the same problem.

The log file [logs.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/1062132/logs.txt)

**Comment by claywilkinson** [comment]
This error: 

```
E/TokenRequestor( 7890): You have wrong OAuth2 related configurations, please check. Detailed error: INVALID_AUDIENCE

E/TokenFragment(  398): Error with silentSignIn: Status{statusCode=DEVELOPER_ERROR, resolution=null}

D/TokenPendingResult(  398):  Calling onResult for callback: GooglePlayGames.Android.TokenResultCallback result: Status: Status{statusCode=DEVELOPER_ERROR, resolution=null} email: <null> id:<null> access: <null>
```

Looks like the web client id you are using is not for a web application that is linked to your game configuration in the Play console.  Can you double check that?

Also, you could try to not request the id token and see if it is successfully authenticating?

**Comment by KevinRM** [comment]
Okey, I checked the client id (Google play console - Game services - My game - Linked apps - select my app - ID client OAuth2) in the Unity3D google setup and it was the same.

On the other hand, I try with RequestIdToken and it failed [logs.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/1062543/logs.txt), then, I try without RequestIdToken and it was successful [logs.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/1062557/logs.txt)

**Comment by claywilkinson** [comment]
Hmm. Can you try on the same device?  It looks like the one requesting the id token and failing is running on a different device than the one that works?  This will just help eliminate any variables that could be hiding the problem.

Also, it still looks like you are not running the code in 0.9.39a.  Can you delete  Assets/GooglePlayGames and Assets/Plugins/Android/MainLibProj and reimport 0.9.39a?

Also can you share the directory listing including file sizes of Assets/Plugins/Android/MainLibProj/libs? (Just in case I am wrong, and you are running the latest code :) )

From the log you are definitely getting authenticated, just the ID token is null.  One of the things 0.9.39a fixes is a race condition that causes this situation.

**Comment by KevinRM** [comment]
Hi, I try in the same device all time, in Samsung S3 with Android 4.3.

I did that when you told me in a previous answer, but I do it again to prevent.

These are the files in libs directory

![captura](https://user-images.githubusercontent.com/10706601/27005271-caf971c6-4e12-11e7-8212-5443f46eb18f.PNG)

Finally, the last logs:
- Without RequestIdToken [logs.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/1066036/logs.txt)
- With RequestIdToken [logs.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/1066039/logs.txt)

**Comment by KevinRM** [comment]
Any help please?

**Comment by KevinRM** [comment]
@claywilkinson 

Hi, after several days looking my app logs, I saw INVALID_AUDIENCE error. Then, I investigated this error and I could solve the problem. I share the solution to help others who may have the same problem.

It occurs by linking the application with Firebase, that this generate a new client ID OAuth 2.0 different of the generated by google developer console. 
So, for solve this and get success on google login and RequestIdToken, re-setup the google plugin in Unity3D and in "Web App Client ID" don't set the ID of Google Play Console, set the generated from Firebase in "https://console.developers.google.com - Manager APIs - your project - credentials" and take from IDs client OAuth 2.0 the ID that their name is "Web client (auto created by Google Service" and their type is Web, then, finish the setup and try the application. 

This solved my problem :D :D

It's very simple to solve but not to realize, since the ID should be the same for Google Play Console as for Firebase if they projects are linked.

Thanks for everything
